arches_modular_reports/src/arches_modular_reports/ModularReport/components/ResourceEditor/ResourceEditor.vue

```diff
@@ -1,46 +1,120 @@
 <script setup lang="ts">
-import { inject, ref, watchEffect } from "vue";
+import { computed, inject, reactive, ref, watchEffect } from "vue";
 
 import Message from "primevue/message";
-import ProgressSpinner from "primevue/progressspinner";
+import Skeleton from "primevue/skeleton";
 
-import { fetchModularReportResource } from "@/arches_modular_reports/ModularReport/api.ts";
-import CardEditor from "@/arches_modular_reports/ModularReport/components/ResourceEditor/components/CardEditor/CardEditor.vue";
 import DataTree from "@/arches_modular_reports/ModularReport/components/ResourceEditor/components/DataTree.vue";
+import GenericCard from "@/arches_component_lab/generics/GenericCard/GenericCard.vue";
+
+import { fetchModularReportResource } from "@/arches_modular_reports/ModularReport/api.ts";
+
+import { EDIT } from "@/arches_component_lab/widgets/constants.ts";
 
-import type { ResourceData } from "@/arches_modular_reports/ModularReport/types.ts";
+import type { Ref } from "vue";
+import type {
+    NodeData,
+    NodegroupData,
+    ResourceData,
+    TileData,
+} from "@/arches_modular_reports/ModularReport/types.ts";
+
+const { selectedNodegroupAlias } = inject("selectedNodegroupAlias") as {
+    selectedNodegroupAlias: Ref<string | null>;
+};
+const { selectedTileId } = inject("selectedTileId") as {
+    selectedTileId: Ref<string | null | undefined>;
+};
 
 const graphSlug = inject<string>("graphSlug")!;
-const resourceId = inject<string>("resourceInstanceId")!;
-const resourceData = ref<ResourceData>();
-const isLoading = ref(true);
+const resourceInstanceId = inject<string>("resourceInstanceId")!;
 
 const emit = defineEmits(["save"]);
 
+const resourceData = reactive<ResourceData>({} as ResourceData);
+const configurationError = ref<Error | null>(null);
+const isLoading = ref(true);
+
 watchEffect(async () => {
     try {
-        resourceData.value = await fetchModularReportResource({
-            graphSlug,
-            resourceId,
-            fillBlanks: true,
-        });
+        Object.assign(
+            resourceData,
+            await fetchModularReportResource({
+                graphSlug,
+                resourceId: resourceInstanceId,
+                fillBlanks: true,
+            }),
+        );
+    } catch (error) {
+        configurationError.value = error as Error;
     } finally {
         isLoading.value = false;
     }
 });
+
+const selectedTileData = computed<TileData | undefined>(() => {
+    const selectedNodegroupAliasedTileData: NodeData | NodegroupData =
+        resourceData.aliased_data[selectedNodegroupAlias.value!];
+
+    if (Array.isArray(selectedNodegroupAliasedTileData)) {
+        return selectedNodegroupAliasedTileData.find(
+            (tileDatum) => tileDatum.tileid === selectedTileId.value,
+        );
```

Comment:
This is going to get interesting when there are more than 1 new cardinality N tiles at play. I'm okay punting this, but let's not lose track of it.

Reply:
In other words, what happens when multiple tiles are in the array with a null tileid.

Reply:
oh -- I thought the empty shape queryset API generated a tileid?

Reply:
I think we debated that a little bit. It's trivial to add it back in. But we removed to so as not to cause confusion with "this tile is already persisted" -- how else can we tell dummy blanks apart from the persisted tiles? Add a `_blank` attribute?

Reply:
oh that's fair 🤔 

Yeah I can see how generating a tileid for a new tile is a confusing pattern.

We're getting around this in querysets by only allowing 1 new tile at a time. We could punt this by enforcing similar constraints at either the resource or nodegroup level. But I'm sure there's _some_ graceful way to handle this.

Regardless I'll file a followup 👍

Reply:
Generating on the backend is nice, it can provide a stronger link between parent and child. So maybe we _do_ want that `auto_generated` attribute.

Reply:
As tiles are edited/saved on the front end, couldn't tile ids be generated then?  It's a front-end solution, but it gets around the whole "how do I tell if a tile was created/edited by the user".